tests/tex/crunch.cpp

```diff
@@ -0,0 +1,112 @@
+#include "./utils.hpp"
+
+#include <btu/common/filesystem.hpp>
+#include <btu/tex/crunch_functions.hpp>
+#include <btu/tex/crunch_texture.hpp>
+
+#include <filesystem>
+
+inline auto load_tex_crunch(const Path &path) -> btu::tex::CrunchTexture
+{
+    auto res = btu::tex::load_crunch(path);
+    INFO(path);
+    REQUIRE(res.has_value());
+    return std::move(res).value();
+}
+
+template<typename Func>
+auto test_expected_crunch(const Path &root, const Path &filename, Func f, Approve approve = Approve::Yes)
```

Comment:
Couldn't this function be merged with test_expected by adding a input path parameter?

Reply:
The problem is not the path, but the `load` vs `load_crunch` functions and their return types (`Result` vs `ResultCrunch`).